metadata/metadata.go

```diff
@@ -229,11 +229,30 @@ func (wrapper SearchWrapper) Set(id ResourceID, res Resource) error {
 	if err := wrapper.ResourceLookup.Set(id, res); err != nil {
 		return err
 	}
-	doc := search.ResourceDoc{
-		Name:    id.Name,
-		Type:    id.Type.String(),
-		Variant: id.Variant,
-	}
+
+	var allTags []string
+	if sourceVariant, ok := res.(*sourceVariantResource); ok {
+        allTags = append(allTags, sourceVariant.serialized.Tags.Tag...)
+    }
+
+	if featureVariant, ok := res.(*featureVariantResource); ok {
+        allTags = append(allTags, featureVariant.serialized.Tags.Tag...)
+    }
+
+	if labelVariant, ok := res.(*labelVariantResource); ok {
+        allTags = append(allTags, labelVariant.serialized.Tags.Tag...)
+    }
+
+	if trainingSetVariant, ok := res.(*trainingSetVariantResource); ok {
+        allTags = append(allTags, trainingSetVariant.serialized.Tags.Tag...)
+    }
```

Comment:
This logic seems like it would be better expressed as a switch statement there will only ever be one match here:

```go
switch res.(type) {
    case *sourceVariantResource:
        allTags = sourceVariant.serialized.Tags.Tag
```

Reply:
fixed here - commit c41c396943af6a33a6475dde4455410c1b4b9fb1